Enhance/#101 implement memo search test

shoutarou123 opened the pull request:
## 対応するissue
<!-- ここに対応するissue番号を書く。issue番号が99なら、「- closes #99」と書く。 -->
- closes #101 検索機能のコントローラーのtestを書く 

## 対応内容
<!-- ここに対応した内容を書く。 -->
- 検索機能のテストを追加しました。ご確認お願いします

shoutarou123 committed:
テストコード追加

shoutarou123 committed:
Merge branch 'main' into enhance/#101-implement-memo-search-test

kuri0616 commented on backend/spec/requests/memos_spec.rb:
```diff
@@ -58,6 +58,63 @@
       end
     end
+
+    context 'あいまい検索する場合' do
```
contextは前提条件や状況を記載するので「あいまい検索する場合」だと少し抽象的かなと思いました！
条件や状況をとしては
クエリパラメータでタイトルが指定された時、コンテンツが指定された時、並び順が指定された時とあるので
contextブロックもその条件ごとに区切る方が意味がわかりやすいかなと思いました(仕様書としての機能もあるため)

たとえばここでいうと
`context 'ログイン中かつ、タイトルが指定された場合'`
のような形です！

kuri0616 commented on backend/spec/requests/memos_spec.rb:
```diff
@@ -58,6 +58,63 @@
+      let!(:first_memo) { create(:memo, title: 'メモタイトル１', content: 'メモコンテンツ１') }
+      let!(:second_memo) { create(:memo, title: 'メモタイトル２', content: 'メモコンテンツ２') }
+
+      before { sign_in(user) }
```
上記2件のデータだけだと、部分一致検索ができている、いないにも関わらず2件ヒットすると思うので
```
      let!(:first_memo) { create(:memo, title: 'メモタイトル１', content: 'メモコンテンツ１') }
      let!(:second_memo) { create(:memo, title: 'メモタイトル２', content: 'メモコンテンツ２') }
      let!(:third_memo) { create(:memo, title: 'その他タイトル', content: 'その他コンテンツ') }
```
と合致しないデータも含めた方が良いかなと思いました！

kuri0616 commented on backend/spec/requests/memos_spec.rb:
```diff
@@ -58,6 +58,63 @@
+        end
+      end
+
+      it 'コンテンツで部分一致するメモが返る' do
+        aggregate_failures do
+          get '/memos', params: { content: 'メモ' }
+          assert_request_schema_confirm
+          assert_response_schema_confirm(200)
+          expect(response.parsed_body['memos'].length).to eq(2)
+          result_memo_ids = response.parsed_body['memos'].pluck('id')
+          expect(result_memo_ids).to contain_exactly(first_memo.id, second_memo.id)
+        end
+      end
```
ここも
context `'ログイン中かつ、コンテンツが指定された場合`
のように条件ごとに切り分けましょう！

kuri0616 commented on backend/spec/requests/memos_spec.rb:
```diff
@@ -58,6 +58,63 @@
+          result_memo_ids = response.parsed_body['memos'].pluck('id')
+          expect(result_memo_ids).to eq([memo_c.id, memo_b.id, memo_a.id])
+        end
+      end
```
ここも同様に昇順の場合、降順の場合でcontextをわけましょう！
例`context 'ログイン中かつ、並び順にdescが指定された場合' `

shoutarou123 committed:
memos_spec.rb修正
describe GET /memosの記述が2つあることによりrubocopの警告を避けるためrubocop:disable RSpec/RepeatedExampleGroupDescription等の記述も追加。

shoutarou123 committed:
githubのrspecエラーになるため記述追加

shoutarou123 committed:
describeを削除し検索テストできるよう修正
